storage: avoid trace use-after-finish

tbg opened the pull request:
action() closed over a context but was then called in a goroutine.
Use the context provided by the stopper instead, which inserts
a span with the proper lifetime.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18627)
<!-- Reviewable:end -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18627#-:-KuVgYtCb_Nel1byhT7B:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: avoid trace use-after-finish

action() closed over a context but was then called in a goroutine.
Use the context provided by the stopper instead, which inserts
a span with the proper lifetime.
